tests/Doctrine/Tests/DBAL/Functional/StatementTest.php

```diff
@@ -349,4 +350,34 @@ public function testExecWithRedundantParameters() : void
         self::expectException(DBALException::class);
         $stmt->execute([null]);
     }
+
+    /**
+     * @throws DBALException
+     *
+     * @dataProvider nonExistingIndexProvider
+     */
+    public function testFetchColumnNonExistingIndex(int $index) : void
+    {
+        if ($this->connection->getWrappedConnection() instanceof PDOConnection) {
+            $this->markTestSkipped('PDO supports this behavior natively but throws a different exception');
```

Comment:
I wouldn't bother handling the PDO exception since the only way to identify it is by the message. It's a logical exception, so it's not supposed to be handled at runtime anyways.